src/globus_sdk/tokenstorage/v2/base.py

```diff
@@ -29,6 +29,19 @@ class TokenStorage(metaclass=abc.ABCMeta):
 
     def __init__(self, namespace: str = "DEFAULT") -> None:
         self.namespace = namespace
+        self._id_token_decoder: globus_sdk.IDTokenDecoder | None = None
+
+    @property
+    def id_token_decoder(self) -> globus_sdk.IDTokenDecoder | None:
+        """
+        An ID Token decoder to use when decoding ``id_token`` JWTs from Globus Auth.
+        By default, a new decoder is used each time decoding is performed.
+        """
+        return self._id_token_decoder
+
+    @id_token_decoder.setter
+    def id_token_decoder(self, value: globus_sdk.IDTokenDecoder) -> None:
+        self._id_token_decoder = value
```

Comment:
Would a plain-jane attribute be simpler than a property?

```suggestion
        # An ID Token decoder to use when decoding ``id_token`` JWTs from Globus Auth.
        # By default, a new decoder is used each time decoding is performed.
        self.id_token_decoder: globus_sdk.IDTokenDecoder | None = None
```

Reply:
Yeah, I noticed this as well after finishing the implementation. Let me chew on it a little.
My reason for keeping it as a property was that the documentation is easier to chunk for sphinx in a property, but maybe that's a bad reason.

Reply:
Okay, why don't I do that! I was hesitant but why not, right?

Reply:
It is, IMO, a bit uglier in the rendered docs, but I can live with it in exchange for saving us some unnecessary method wrappers.